### ISSUE TITLE
Download links from Testing UI do not resolve if accessing Testrun from different IP addresses/hostnames

### ISSUE THREAD

**Issue by duncangreene** [google/testrun#892]
**Describe the bug**
The download links in the Testing UI do not resolve if accessing Testrun via different IP addresses/hostnames.

In this case I have a machine 192.168.0.67 running Testrun, I am web browsing to it from another machine 192.168.0.55. All Testrun browsing is working okay, with the exception of the download links in the Testing window, that appear to not be relative URLs, and instead try to resolve against localhost.

**To Reproduce**
1. Web browse to Testrun from an external machine (e.g. http://192.168.0.67:8080/).
2. Run a test and let it complete. Cancel out of the report download modal window once the test has completed.
3. Attempt to Download PDF or ZIP file as below.
![Image](https://github.com/user-attachments/assets/143a6e58-81bc-4879-837e-665508baca7c)
4. Error as below.
![Image](https://github.com/user-attachments/assets/7ed0f905-a865-4794-ab2f-19f08cc60679)

**Expected behavior**
Download link should resolve to a relative URL, to cater for accessing Testrun via hostname/IP address/localhost/behind NAT/etc.

**Environment (please provide the following information about your setup):**
 - OS: Ubuntu
 - Version 24.04.1
 - Testrun 2.0

**Comment by jhughesbiot** [comment]
This requires a little extra configuration that does not appear to be documented or available in the UI at this time.

Navigate to <Tesrun Install Directory>/local/system.json and add the variable api_url and point it to your systems IP address.

Example system.json with this option included:

```
{
  "network": {
    "device_intf": "eth0",
    "internet_intf": "eth1"
  },
  "log_level": "DEBUG",
  "startup_timeout": 60,
  "monitor_period": 60,
  "max_device_reports": 5,
  "api_url": "http://10.0.0.5",
  "api_port": 8000,
  "org_name": "",
  "single_intf": false
}
```

This would make all API routes go to 10.0.0.05 instead of localhost when connecting via the UI.

**Comment by duncangreene** [comment]
Good to know about the `api_url` workaround. 👍

Is there scope to add a feature request for this one to get those particular hyperlinks on the UI working relatively? The rest of the hyperlinks across the UI work perfectly, and it'd be nice to be able to decouple Testrun from being dependent on what hostname/IP address a sysadmin has configured the Testrun machine with? 🤞

**Comment by jboddey** [comment]
Thanks for the bug report. I believe this has been a known issue for some time hence the implementation of api_url. I will see if we can add this setting to the UI.

**Comment by jboddey** [comment]
This issue is currently being addressed. A PR is expected soon.

**Comment by jboddey** [comment]
Fixed in PR #954. This will be released shortly in v2.1
